### PR TITLE
shared-mime-info: Requires perl during build

### PR DESCRIPTION
**Pull request by dholm**

### CODE REVIEW

File: Formula/shared-mime-info.rb [hunk 1]
```diff
@@ -24,6 +24,7 @@ class SharedMimeInfo < Formula
   depends_on "intltool" => :build
   depends_on "gettext"
   depends_on "glib"
+  depends_on "perl" => :build
```
Comment (sjackman) [inline Formula/shared-mime-info.rb]: Please change to
```ruby
depends_on :perl => :build unless OS.mac?
```

**Comment by sjackman** [comment]
Which distribution of Linux are you using that does not have Perl installed?

File: Formula/shared-mime-info.rb (hunk 1 above)
Comment (dholm) [inline Formula/shared-mime-info.rb]: @sjackman Done!

**Comment by dholm** [comment]
@sjackman RHES 6.8. It comes with perl of course but I think it was too old (5.10.1).

**Comment by sjackman** [comment]
Please post `brew gist-logs shared-mime-info` of the error that you were encountering. That way searching for the error message will yield this PR, in case of a regression.

**Comment by dholm** [comment]
Ok, so I unlinked perl from Linuxbrew, removed shared-mime-info and intltool (because intltool was using my Linuxbrew perl). Now I get the following error when installing:

```
$ brew install -v shared-mime-info
intltool: Unsatisfied dependency: XML::Parser
Homebrew does not provide special Perl dependencies; install with:
  `cpan -i XML::Parser`
Error: An unsatisfied requirement failed this build.
```

Even though I just did install `XML::Parser` using my system provided perl.

```
$ perldoc -l XML::Simple
/app/perl/5.12.3/RHEL64/lib/XML/Simple.pm
```

**Comment by dholm** [comment]
After getting intltool to work shared-mime-info fails with the same error as intltool in #2659.

**Comment by sjackman** [comment]
Try either
```
sudo apt-get install -y libxml-parser-perl
```
or
```
brew install perl cpanm
cpanm XML::Parser
```

**Comment by dholm** [comment]
@sjackman I don't have root access so I cannot deploy anything globally.  The second method works but only if the package depends on the brewed perl.  If I install `XML::Parser` locally with `cpanm` using the system's perl `shared-mime-info` still fails.

**Comment by sjackman** [comment]
> The second method works but only if the package depends on the brewed perl.  If I install `XML::Parser` locally with `cpanm` using the system's perl `shared-mime-info` still fails.

Hmm. Yes. That hadn't occurred to me until you mentioned it. The brewed `python` won't be in the `PATH` unless the formula depends on `python`.

**Pull request closed without merging**